File: api/app/schemas/job.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

from app.schemas.company import CompanyRead


class JobBase(BaseModel):
    title: str
    location: Optional[str] = None
    employment_type: Optional[str] = None
    job_url: str
    external_id: Optional[str] = None


class JobCreate(JobBase):
    company_id: int


class JobRead(JobBase):
    id: int
    company_id: int
    is_active: bool
    first_seen_at: datetime
    last_seen_at: datetime
    company: CompanyRead

    model_config = {"from_attributes": True}


class PaginatedJobs(BaseModel):
    items: list[JobRead]
    total: int
    page: int
    size: int
    pages: int

File: client/src/types/job.ts
import type { Company } from "./company";

export interface Job {
  id: number;
  external_id: string | null;
  title: string;
  location: string | null;
  employment_type: string | null;
  job_url: string;
  company_id: number;
  is_active: boolean;
  first_seen_at: string;
  last_seen_at: string;
  company: Company;
}

export interface JobCreate {
  title: string;
  location?: string;
  employment_type?: string;
  job_url: string;
  external_id?: string;
  company_id: number;
}

export interface PaginatedJobs {
  items: Job[];
  total: number;
  page: number;
  size: number;
  pages: number;
}

export interface JobFilters {
  search?: string;
  location?: string;
  company_id?: number;
  page?: number;
  size?: number;
}
